libvirt/image/Makefile

```diff
@@ -68,17 +69,42 @@ STATIC_LIB_BUILDER = ../../../kata-containers/ci/install_libseccomp.sh
 STATIC_LIB_DIR     = $(abspath staticlib)
 STATIC_LIB         = $(STATIC_LIB_DIR)/kata-libseccomp/lib/libseccomp.a
 
-build: $(IMAGE_FILE)
+build: ubuntu-build
+ubuntu-build: $(UBUNTU_IMAGE_FILE)
 
-$(IMAGE_FILE): $(BINARIES) $(FILES)
+rhel-build: $(RHEL_IMAGE_FILE)
+
+checkvars:
+ifndef RHEL_IMAGE_URL
+	$(error "RHEL_IMAGE_URL is not defined")
+endif
+ifndef RHEL_IMAGE_CHECKSUM
+	$(error "RHEL_IMAGE_CHECKSUM is not defined")
+endif
+
+
+$(RHEL_IMAGE_FILE): | checkvars $(BINARIES) $(FILES)
+	rm -fr output
+	rm -f cloud-init.img
+	cloud-localds cloud-init.img userdata.cfg
+	mkdir -p toupload
+	packer build -var qemu_image_name=${RHEL_IMAGE_FILE} \
+		-var cloud_image_url=${RHEL_IMAGE_URL} \
+		-var disk_size=11144 \
+		-var ssh_username=cloud-user \
+		-var cloud_image_checksum=${RHEL_IMAGE_CHECKSUM} -only=qemu.rhel .
+	rm -fr toupload
+	rm -f cloud-init.img
+
+$(UBUNTU_IMAGE_FILE): $(BINARIES) $(FILES)
 	echo ${UBUNTU_IMAGE_CHECKSUM}
 	rm -fr output
 	rm -f cloud-init.img
 	cloud-localds cloud-init.img userdata.cfg
 	mkdir -p toupload
-	packer build -var qemu_image_name=${IMAGE_FILE} \
+	packer build -var qemu_image_name=${UBUNTU_IMAGE_FILE} \
 		-var cloud_image_url=${UBUNTU_IMAGE_URL} \
-		-var cloud_image_checksum=${UBUNTU_IMAGE_CHECKSUM} .
+		-var cloud_image_checksum=${UBUNTU_IMAGE_CHECKSUM} -only=qemu.ubuntu .
```

Comment:
how about still using the same target but using the template file based on provided option

```
packer build ...opts.. $TEMPLATE
```

Default TEMPLATE set to qemu-ubuntu.pkr.hcl